repos/system_upgrade/common/actors/checkfirstpartitionoffset/libraries/check_first_partition_offset.py

```diff
@@ -0,0 +1,64 @@
+from leapp import reporting
+from leapp.libraries.common import grub
+from leapp.libraries.stdlib import api, CalledProcessError, run
+
+SAFE_OFFSET_kiB = 1024  # 1MiB
+
+
+def is_first_partition_offset_ok(device):
+    try:
+        partition_table = run(['parted', device, 'unit', 'kiB', 'print'], split=True)['stdout']
+    except CalledProcessError:
+        # Unlikely - if the disk has no partition table, `parted` terminates with 0 (no err). Parted exits with an err
+        # when the device does not exists, or if it is too small to contain a partition table.
+        api.current_logger().error('Failed to run `parted` to obtain partition table of the device.')
+        return True  # There is no first partition, so the predicate holds vacuously
+
+    table_iter = iter(partition_table)
```

Comment:
This is a neat idiom!  I hadn't seen this before but I can see the advantages to using the iterator.  Nice work.

Reply:
Thank you for your kind words :bow: